### ISSUE TITLE
Multiple Order Confirmation Emails

### ISSUE THREAD

**Issue by Swiftless** [zipMoney/magento2#72]
When completing a purchase in Magento 2.3.3 or Magento 2.2.7, customers are sent two order confirmation emails every single time.

Order history in Magento Admin Dashboard shows the following for every order. NOTE the two "Customer Notified" before and after the payment entry.

> 29 Jan 2020 12:21:14 pm Pending Customer Notified
> Notified customer about invoice #5201003181DA.
> 29 Jan 2020 12:21:14 pm Pending Customer Not Notified
> Registered notification about captured amount of $15.94. Transaction ID: "ch_FIY7OOCxS4suAkYVyuUvH1"
> 29 Jan 2020 12:21:13 pm Pending Customer Notified

**Comment by benzhangzip** [comment]
Hi @Swiftless,

We could confirm that we had the similar behaviors like yours, there is 2 notification being sent to customer, but it is not duplicate, one is "Order Confirmation" email, and one is "Order Invoice" email.
However, most Magento developer or business will like to simplify the email and make the "Order Confirmation" = "Order Invoice" and try to skip the invoice email, which is understandable as 95% user will charge directly and order confirmation is invoice at that moment.

![image](https://user-images.githubusercontent.com/42924127/73699084-f4daaf80-4736-11ea-98b6-fd57e56641c0.png)

As you could see from the order status history customer notified in the order creation and Zip API charge happened later, then log history to show you the charge id in status history (no email sent) then try to notify the customer via email in the third status history update which is invoice email, it is useful for merchants doing auth and capture, but not really useful for capture directly merchants.

If you really want to disable invoice email sending, you could comment out some code in the file:
/vendor/zipmoney/magento2/Model/Charge.php  on about line 433
`// Invoice
    $invoice = $payment->getCreatedInvoice();
    
    if ($invoice) { 
      if ($this->_order->getCanSendNewEmailFlag()) {
        try {
          //$this->_orderSender->send($this->_order);
        } catch (\Exception $e) {
          $this->_logger->critical($e);
        }
      }   

      $this->_order->addStatusHistoryComment($this->_helper->__('Notified customer about invoice #%s.', $invoice->getIncrementId()))
                   ->setIsCustomerNotified(true);
      
      $this->_orderRepository->save($this->_order);                    
    }`

commented out the "$this->_orderSender->send($this->_order);" then it will skip the invoice email sending. Hope this helps.

**Comment by Swiftless** [comment]
Hi @benzhangzip thanks for the response.

I don't think the reason is correct. 

I have different email templates for Order and Invoice and it's the Order template that I receive twice. I still receive duplicate Order emails after disabling the Invoice email.

If I disable the Order email I don't get any emails at all.

From what I understand and what you've said, there should be two different templates being sent to me, the Order template and the Invoice template but that is not the case.

**Comment by benzhangzip** [comment]
Hi @Swiftless,

Did you try to comment out the code like I said? it is the one that causing the order confirmation email being sent twice. 
Just comment out the code in
/vendor/zipmoney/magento2/Model/Charge.php on about line 433
just remove or comment out this code
```
$this->_orderSender->send($this->_order);
```
It is using a wrong email sender of order
```
\Magento\Sales\Model\Order\Email\Sender\OrderSender
```
But it should be invoice email sender (also checking sending or not condition is wrong)
```
Magento\Sales\Model\Order\Email\Sender\InvoiceSender
```
But unfortunately our developer when developed 2 years ago did not consider that they should not hard code send order notification email as default Magento place order flow will send email anyway when order created, so just comment out this function then it will not send duplicate order confirmation emails. Please try that out and let us know if that fix the issue.

Thanks

**Comment by Swiftless** [comment]
Hi @benzhangzip,

It looks like we're on the same page now with why this is happening and it is definitely a problem with the extension.

Commenting out that line of code should fix the issue, I'll pass it on to our implementation team for testing.

**Comment by suresh18th** [comment]
I am facing same issue on Magento 2.3.5-p1.
I tested issue by installing extension on fresh Magento instance.
Please advise.

**Comment by benzhangzip** [comment]
Hi @suresh18th,
Could you provide us more details please? Such as zipMoney logs and Magento2 system logs, We recently tested in Magento2.4 and all working fine. Is that installation related? Have you run DB update command when install, even we did not introduce new table or alter magento default tables, DB update command still need to be run in order to save extension version in the module table when first time install Zip.

**Comment by suresh18th** [comment]
Hi,

After getting issue on our staging instance, I tested on fresh Magento instance by following steps.

1) I installed the Magento fresh instance 2.3.5-p1. URL of this fresh instance is http://m2fresh.demolocations.com/.
2) I added some products for testing purpose.
3) I installed zipmoney extension.
4) I enable it by running following command
	php bin/magento module:enable ZipMoney_ZipMoneyPayment
5) After this i run following command
rm -rf var/cache/* var/page_cache/* var/generation/* var/di pub/static/* var/view_preprocessed/* generated/*
php bin/magento setup:upgrade
php bin/magento setup:di:compile
php bin/magento setup:static-content:deploy  -f
php bin/magento cache:clean
php bin/magento cache:flush

6) After above steps we placed order on http://m2fresh.demolocations.com/ and got 2 order confirmation email.

7) We tested this issue on 3 instances (1. Staging instance, 2. local development instance, 3. fresh Magento instance http://m2fresh.demolocations.com/).
We are getting this issue on all the instances.
It can be easily replicate by installing fresh magento instance 2.3.5-p1 and installing zipMoneyPayment extension on this magento instance.

[zipMoneyPayment.log](https://github.com/zipMoney/magento2/files/5195287/zipMoneyPayment.log)
